Caltech-IPAC/firefly
# DM-5792: Support artifacts on all images that have them (currently wise and 2mass)

- Image FileRetrievers can add artifact to RelatedData
  - WISE and 2MASS supported
  - Artifact drawing layer uses lazy loading
  - added layer title matching for visibility, color, and symbol changing
  - firefly: IbeQueryArtifact replaces ife: FinderChartQueryArtifact
  - Also added support for getting adding RelatedData artifacts in IbeFileRetrieve

_The big concept:_

- When an image plot is request is sent to the server it not only returns the image data but also how to retrieve "related data". 
- Sometime this related data are image mask in other cases it is artifact data.
- What related data is return is query specific.  This way the client does not have to know the details of the project.
- A RelatedData object can be turned into a server search request to retrieve it when the user request to see it.
- There are so far 3 types of related data: mask, tabular (artifacts), and image (not yet implemented).
- The mask related data was implemented months ago.  This pull request implements the tabular related data.



_to test:_
 - bring up the firefly or irsa viewer, or finder chart, or time series.
 - when you have a wise or 2mass layer up you should see the artifact drawing layers on the dialog.  They are off by default.
  - if you have more than one image such as several wise or several 2mass images. Checking and unchecking artifact turn all of them of the same type on and off.
